rllib/utils/actor_manager.py

```diff
@@ -442,6 +443,9 @@ def __fetch_result(
             remote_calls: list of remote calls to fetch.
             timeout_seconds: timeout for the ray.wait() call. Default is None.
             return_obj_refs: whether to return ObjectRef instead of actual results.
+            mark_healthy: whether to mark certain actors healthy based on the results
```

Comment:
**q**: why don't we do this by default? Why not mark actors as healthy if they are marked as not healthy and they have been successful?

Reply:
yeah, sort of explained this in the docstring.
we want to automatically mark unhealthy, but not automatically mark healthy, because workers usually need to have states synced before they can put back into work.
notice that for IMPALA's aggregator workers, which are stateless, we mark healthy directly with the probe calls because of this.